test/lint/lint-python-utf8-encoding.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Make sure we explicitly open all text files using UTF-8 encoding to avoid
+# potential issues on the BSDs where the locale is not always set.
+
+EXIT_CODE=0
+OUTPUT=$(git grep -i " open(" -- "*.py" | grep -vE 'encoding=.(utf8|utf-8).' | grep -vE "open\([^,]*, ['\"][^'\"]*b[^'\"]*['\"]")
+if [[ ${OUTPUT} != "" ]]; then
+    echo "Python's open(...) seems to be used to open text files without explicitly"
+    echo "specifying 'encoding=utf8':"
```

Comment:
Should say encoding='utf8'? Also, you might want to use the single quotes(') when patching the python files in this pull, since this seems the more common way for short strings. Just a nit, though.